### PR TITLE
flavors added to foods and drinks DM, and taste for reagents organics. and ethanol

### PR DESCRIPTION
**Pull request by fernandos13**
## About The Pull Request
flavors added to foods and drinks DM
adds flavors to a bunch of food and tweak some drinks
also adds new favor profiles to drink
								 TASTE_SLIMEY
								 TASTE_REFRESHING
								 TASTE_DRY
and to food it adds
									SALTY_FOOD
									SPICY_FOOD
									FLOURY_FOOD
									UMAMI_FOOD

## Why It's Good For The Game
fixes some bugs. and give players more options

## Changelog
:cl:Fernandos33
add: TASTE for some foods
add: NEW tastes to roll in insights
tweak: Drinks got a more matching taste profile aswell as new taste profiles
tweak: thanks to Evankhell561 all reagents can now have a taste too.
/:cl:

### CODE REVIEW

**Comment by fernandos13** [comment]
hold up need to add some snack flavors.

**Comment by fernandos13** [comment]
time to iron this out

File: code/modules/reagents/reagent_containers/food/snacks.dm
Comment (Evankhell561) [inline code/modules/reagents/reagent_containers/food/snacks.dm:378]: You cannot put food labels on reagents. If you are going to do it you will have to work a lot more than this. At the moment only the ethanol subtypes.

Comment (fernandos13) [reply code/modules/reagents/reagent_containers/food/snacks.dm:378]: oh wait. only ethanol.. why? 
can you point me the root of it. so i can get it to work with reagents

Comment (fernandos13) [reply code/modules/reagents/reagent_containers/food/snacks.dm:378]: but WAIT
at the line "23"
the /obj/item/weapon/reagent_containers/food/snacks
have a var for taste tag

and that was put there by yourself

Comment (fernandos13) [reply code/modules/reagents/reagent_containers/food/snacks.dm:378]: but WAIT
at the line "23"
the /obj/item/weapon/reagent_containers/food/snacks
have a var for taste "var/list/taste_tag = list(BLAND_FOOD)"

and that was put there by yourself.  (i'm a shit coder so i'm very confused)

Comment (Evankhell561) [reply code/modules/reagents/reagent_containers/food/snacks.dm:378]: forgive me, I must rest and I am seeing things that are not.

Comment (fernandos13) [reply code/modules/reagents/reagent_containers/food/snacks.dm:378]: its ok. your point was helpfull nonetheless and also the points made in discord. i thank you a lot for it

**Comment by fernandos13** [comment]
do not merge its not fixed but some help would be welcome on this thing. happens to me a lot of times when trying to set up procs and i don't get why

File: code/modules/reagents/reagents/food-Drinks.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents/food-Drinks.dm:118]: The only problem with this is if people start taking salt to satisfy their need for a salty drink.

Comment (fernandos13) [reply code/modules/reagents/reagents/food-Drinks.dm:118]: hummm i see. on a note people already take 120u of sugar around moebius for sanity
. lemme fix this salty thing tho

**Comment by fernandos13** [comment]
this was a joke^^^^. you got a good point changes made

File: code/modules/reagents/reagents/food-Drinks.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents/food-Drinks.dm]: return this label, and add below this line this:
sanity_gain_ingest = 0

File: code/modules/sanity/sanity_mob.dm
Comment (Evankhell561) [inline code/modules/sanity/sanity_mob.dm:278]: change this line to:
`if(resting && sanity_gain_ingest  &&E.taste_tag.len)
`

File: code/modules/reagents/reagents/food-Drinks.dm
Comment (Evankhell561) [reply code/modules/reagents/reagents/food-Drinks.dm]: also add
sanity_gain_ingest = 0 to all the condiments and things we don't want to provide "rest".

File: code/modules/sanity/sanity_mob.dm
Comment (Evankhell561) [reply code/modules/sanity/sanity_mob.dm:278]: or better:
```
/datum/sanity/proc/onReagent(datum/reagent/E, multiplier)
	if(E.taste_tag.len)
		changeLevel(E.sanity_gain_ingest * multiplier)
		if(resting)
			for(var/taste_tag in E.taste_tag)
				add_rest(taste_tag, 4 * multiplier/E.taste_tag.len)
```

File: code/modules/reagents/reagents.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents.dm:42]: Set this to zero by default and add sanity_gain_ingest = 0.5 to the types or subtypes of reagents that interest us.

Comment (fernandos13) [reply code/modules/reagents/reagents.dm:42]: sure
 can do

**Comment by fernandos13** [comment]
this pr got mixed up somehow with NT pistol lemme double check if i managed to clean it

**Comment by fernandos13** [comment]
NOW CLEAN and good to go again

**Comment by fernandos13** [comment]
can do. just need some time to compile and run. for some reason hosting on byond takes AGES for me. not sure if its like that for everyone

File: code/modules/reagents/reagents.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents.dm:155]: change this line to:
`/datum/reagent/proc/affect_ingest(mob/living/carbon/M, alien, effect_multiplier)`

Along with all the procs for the files you are changing. Remove the `var/` used as arguments in the procs.

File: code/modules/reagents/reagents.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents.dm:162]: for example, this:
/datum/reagent/proc/affect_touch(mob/living/carbon/M, alien, effect_multiplier)

File: code/modules/reagents/reagents.dm
Comment (fernandos13) [reply code/modules/reagents/reagents.dm:155]: oh ok. i didn't even tried to mess with that since it was already there and i didn't wanted to mess it up. will do

**Comment by Evankhell561** [comment]
> can do. just need some time to compile and run. for some reason hosting on byond takes AGES for me. not sure if its like that for everyone


On my previous computer it took me 5 minutes to host.

**Comment by fernandos13** [comment]
whats bugging me now is. i literally coppied onAlcohol and made it onReagent. whats wrong with it

**Comment by fernandos13** [comment]
this and that. all put in the conforms
if(multiplier <= 1)
and
You shouldn't delete this: resting

**Comment by TorinoFermic** [comment]
Last thing, there's merge conflicts at this moment.

File: code/modules/reagents/reagents.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents.dm:163]: Var/alien to alien

File: code/modules/reagents/reagents.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents.dm:159]: These three lines of code repeats too much. it would be nice to make it a proc..

**Comment by fernandos13** [comment]
huh.. its a proc calling a proc but here it goes

done

**Comment by fernandos13** [comment]
wait there's a problem(but since travis is not here it can't be spotted i'm fixing it)

File: code/modules/reagents/reagents.dm
Comment (Evankhell561) [inline code/modules/reagents/reagents.dm:152]: mob/living/carbon/human/H

Comment (fernandos13) [reply code/modules/reagents/reagents.dm:152]: sure. i'll make other changes to support it too

Comment (fernandos13) [reply code/modules/reagents/reagents.dm]: done

**Review by Evankhell561 — DISMISSED** [review]
merge conflicts

File: code/modules/reagents/reagent_containers/food/snacks.dm
Comment (fernandos13) [inline code/modules/reagents/reagent_containers/food/snacks.dm:2861]: conflict and cheese fixed

**Comment by Evankhell561** [comment]
For reasons beyond my control I cannot test this pr, so I cannot fully approve it but someone should review this.

File: code/modules/reagents/reagents/food-Drinks.dm
Comment (TheShown911) [inline code/modules/reagents/reagents/food-Drinks.dm:951]: Is this intended?

Comment (fernandos13) [reply code/modules/reagents/reagents/food-Drinks.dm:951]: yes, doctor gib have over 48 flavors its said. altough, its set as a "junk food" so it shouldn't work for insight

Comment (fernandos13) [reply code/modules/reagents/reagents/food-Drinks.dm:951]: A delicious blend of 42 different flavours

Comment (fernandos13) [reply code/modules/reagents/reagents/food-Drinks.dm:951]: desc says. A delicious blend of 42 different flavours

 so i followed it

Comment (TheShown911) [reply code/modules/reagents/reagents/food-Drinks.dm:951]: Based

**Review by TheShown911 — APPROVED** [review]

File: code/modules/reagents/reagents/food-Drinks.dm
Comment (GrayRachnid) [reply code/modules/reagents/reagents/food-Drinks.dm:951]: This might be a bit unbalanced as these can be bought at any soda machine.

Comment (fernandos13) [reply code/modules/reagents/reagents/food-Drinks.dm:951]: this gave a idea how about making the can price 1k? might be over killing it. but still how about it

**Pull request merged**
